### DISCUSSION TITLE
Will the old postcode-based API be updated to the new formats?

### DISCUSSION THREAD

**Discussion by acstacey** [amberelectric/public-api#73] [Q&A] [answered]
I have been using the https://api.amberelectric.com.au/prices/listprices endpoint and the formula to calculate usage and feed-in prices.  Recently that API has been very unreliable with sometimes not returning any recent actual data and only forecasts.

I have been using these formulae to calculate the prices and they don't match up at all with the prices shown in the Amber app
From the data there, the formula for usage prices is:
data.staticPrices.E1.totalfixedKWHPrice + data.staticPrices.E1.lossFactor * data.variablePricesAndRenewables.[period].wholesaleKWHPrice
And the formula for Export to Grid prices is the same:
data.staticPrices.B1.totalfixedKWHPrice + data.staticPrices.B1.lossFactor * data.variablePricesAndRenewables.[period].wholesaleKWHPrice

**Comment by madpilot** [comment, marked as answer]
No, it's no longer supported and will be turned off in the new year. 

Please use the new API.

**Reply by acstacey** [reply]
Thanks

Are there any plans to add query by location, or will it only be possible to get prices for specific sites from now on?

**Reply by madpilot** [reply]
All queries will be by site - we aren't able to accurately price a location without knowing specifics about the site's tariffs. This is one of the reasons the old API was retired.

**Reply by acstacey** [reply]
Makes sense :)
Cheers

**Reply by madpilot** [reply]
https://www.nemy.io/ has done a IFTTT integration that will give you the wholesaleKWHPrice part of that equation. You can work out the rest of the formula by looking up your current network prices, but if you are an Amber customer the new API does all the heavy lifting for you.

**Reply by alexinthis** [reply]
Looks like it may have been turned off a little early, it has stopped working for me over night.

**Reply by madpilot** [reply]
I just tried it and it's still working.

**Reply by alexinthis** [reply]
> I just tried it and it's still working.

It does still seem to be returning vales however I have been receiving the same data since 10:02pm last night (21/12/21). For my postcode 2103 my vales are:

logTime: "2021-12-22T10:09:28"
logKWhPrice: "32.997800000000005"
retailKWhPrice: 45.898834575424004

**Reply by madpilot** [reply]
Is there any reason you can't use the new API? We aren't turning off the old API for a bit, but we won't fix it either. What is your use-case?

**Reply by alexinthis** [reply]
Oh I can, I wasn't aware there was a new one until I started looking into why my current one stopped working today. My use case is displaying information on my iPad to view current pricing in the kitchen. I call the API from Node Red and send the data to InfluxDB to view in Grafana. I also send it to HomeAssistant at the same time for the display. 

I'll move over to the new API. 

Can I just ask a simple question while you're reading this. I am just missing something here, I am not great with API's. I use HTTP requests to receive the data, I didn't use to need the authorise, can you show me an http example of calling /sites with the API token in it?

**Reply by madpilot** [reply]
Yep! Just set a a header called "Authorization" with a value of "Bearer [whatever-your-token-is]"

It's been a while since I've used Node Red, but if I remember correctly, the HTTP node has the option to set custom headers

**Reply by madpilot** [reply]
Also: We have a native home assistant integration: https://www.home-assistant.io/integrations/amberelectric

**Reply by alexinthis** [reply]
Thanks, I'll try that. 

I did just mean something I can put in a web browser to test though if that's simple for you to explain?

**Reply by madpilot** [reply]
There are cURL examples in [the documentation](https://app.amber.com.au/developers)

**Reply by alexinthis** [reply]
> Also: We have a native home assistant integration: https://www.home-assistant.io/integrations/amberelectric

I just tried this, using my API key it does not work, I receive a red ! where it asks for the API Key and no further info, it looks like there is space for an error message but nothing is there. 

![image](https://user-images.githubusercontent.com/56566006/147016202-0669ad1f-49b6-426c-ae53-52d3f6a53352.png)

**Reply by madpilot** [reply]
And you can actually run queries from there too

**Reply by madpilot** [reply]
> > Also: We have a native home assistant integration: https://www.home-assistant.io/integrations/amberelectric
> 
> I just tried this, using my API key it does not work, I receive a red ! where it asks for the API Key and no further info, it looks like there is space for an error message but nothing is there.
> 
> ![image](https://user-images.githubusercontent.com/56566006/147016202-0669ad1f-49b6-426c-ae53-52d3f6a53352.png)

Weird. I wonder if something has changed recently - what version of HA are you running? I'll see if I can duplicate locally

**Reply by madpilot** [reply]
Ah. You're account is closed (assuming I'm looking at the right Alex Gregory) - so the API will return an empty set of sites, and the home assistant plugin won't work. Indeed, the new API won't work either.

**Reply by madpilot** [reply]
If that isn't right - email info@amber.com.au and our accounts teams can look in to it for you

**Reply by alexinthis** [reply]
> Ah. You're account is closed (assuming I'm looking at the right Alex Gregory) - so the API will return an empty set of sites, and the home assistant plugin won't work. Indeed, the new API won't work either.

That would explain it. I went to sign up back in May 2021 and just I was in that 10 day waiting period the prices were spiking like crazy so  I backed out, it really did settle down and I should have stayed with it. Since then I have always had access to the site and pricing which surprised me. 

This is what it looked like for me at the time:
![image](https://user-images.githubusercontent.com/56566006/147017053-19fb4181-981e-4d24-9147-378d997d0930.png)

Thanks for all your help, I'll check back in once I'm a customer in a month or so.

**Comment by madpilot** [comment]
@acstacey Can you give me some more info about your use-case? We've been getting a number of requests for location based search, and I want to understand why.

**Reply by acstacey** [reply]
I was putting together a dashboard for people to be able to monitor electricity prices by region
